plutus-tx/src/PlutusTx/Builtins.hs

```diff
@@ -69,7 +70,7 @@ module PlutusTx.Builtins (
                                 , toBuiltin
                                 ) where
 
-import           Prelude                    hiding (String, error)
+import           Prelude                    (Bool (..), Integer, const, uncurry)
```

Comment:
should we be using the ones from `PlutusTx.Base`?

Reply:
Maybe we should qualify all uses of the Haskell prelude just to make it really obvious where we're using it?

Reply:
I like the idea.

Should we then create a simple `PlutusTx.Integer` module that just exports `Integer`? I dislike a bit that there are types that are Haskell and Plutus one simultaneously.

Reply:
Yeah, let's do that. I increasingly prefer the style where we act "as if" the on-chain and off-chain types are different, even if they are in fact the same. See e.g. the builtins stuff where we sort-of pretend that there's a builtin integer type that's different.